### PR TITLE
Trivy scan on pr

### PR DESCRIPTION
**Pull request by npolshakova**
# Description

Automatic firing of CVEs should now create alert in code scanning.

# Context

https://github.com/solo-io/gloo/issues/4542

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4542

### CODE REVIEW

File: .github/workflows/trivy-analysis-scheduled.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: security-scan-scheduled
```
Comment (sam-heilbron) [inline .github/workflows/trivy-analysis-scheduled.yaml:1]: Is there anything stopping us from using the same workflow for both (the scheduled and on release) and just support both types of triggers? Do these two run the same operations?

File: .github/workflows/trivy-analysis-release.yaml [hunk 2]
```diff
@@ -39,18 +39,6 @@ jobs:
           VERSION=$(git describe --tags | cut -c 2-)
           TAGGED_VERSION=v$(echo $VERSION) make ${{ matrix.image-type }}-docker
           echo "::set-output name=VERSION::$(echo $VERSION)"
-      - name: Run Trivy vulnerability scanner
```
Comment (sam-heilbron) [inline .github/workflows/trivy-analysis-release.yaml:42]: Why do we remove this?

File: .github/workflows/trivy-analysis-scheduled.yaml (hunk 1 above)
Comment (npolshakova) [reply .github/workflows/trivy-analysis-scheduled.yaml:1]: If we combine both, the scheduled job would also create the security update docs.

File: .github/workflows/trivy-analysis-release.yaml (hunk 2 above)
Comment (npolshakova) [reply .github/workflows/trivy-analysis-release.yaml:42]: This got moved to the scheduled scan, before we were running one scan that output the sarif and one that output the markdown on releases. Now the markdown scan report is only getting run on release.

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**